codegen/templates.go

```diff
@@ -0,0 +1,19 @@
+package codegen
+
+type templateType int
+
+const (
+	templateTypeDataSource templateType = iota
+	templateTypeResource
+	templateTypeDoc
+)
+
+func (t templateType) String() string {
+	switch t {
+	case templateTypeDataSource:
+		return "datasources"
+	case templateTypeResource:
+		return "resources"
+	}
+	return "docs"
```

Comment:
Since the `templateType` is ultimately an integer that could be set to any value, I recommend adding a case for `templateTypeDoc` that returns `"docs"` and having a default value - something like `"unrecognizedtype"` or something like that. That way we can easily distinguish files that were created from bad `templateType`s.

Reply:
Sure, I can do that.

Reply:
https://github.com/terraform-providers/terraform-provider-vault/pull/738/commits/0817194e75a179dfda560319faf19fe0ed2499c3

I made it the first option, corresponding to a value of 0, because of it being the default/unset value for an integer.